packages/create-actor/src/macros.ts

```diff
@@ -0,0 +1,67 @@
+import { readFile, readdir, stat } from "node:fs/promises";
+import path from "node:path";
+import micromatch from "micromatch";
+import pkgJson from "../package.json" with { type: "json" };
+import { PLATFORM_SLUGS } from "./platforms";
+
+const EXAMPLES_PATH = path.join(__dirname, "../../../examples");
+
+const IGNORED_PATHS = /platforms|benches|tsconfig.json/;
+
+interface ExamplesRegistry {
+	[key: string]: {
+		name: string;
+		slug: string;
+		supports: string[];
+		files: Record<string, string>;
+	};
+}
+
+export async function getExamples(): Promise<ExamplesRegistry> {
+	const registry: ExamplesRegistry = {};
+
+	const dirs = await readdir(EXAMPLES_PATH, { encoding: "utf-8" });
+
+	for (const dir of dirs) {
+		const files = await readdir(path.join(EXAMPLES_PATH, dir), {
```

Comment:
this should respect gitignore. i think you can use `git ls-files` to do this, might need to mess with the command.

this should also remove the need for IGNORED_PATHS